set.mm

```diff
@@ -537332,49 +537345,156 @@ singleton on a couple (with disjoint domain) at a point in the domain
 
   A few basic theorems to start affine, Euclidean, and Cartesian geometry.
 
+  The first step is to define real vector spaces, then barycentric coordinates
+  and convex hulls.
+
 $)
 
 
 $(
 -.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
-  Convex hull in real vector spaces
+  Real vector spaces
 -.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
 
-  A few basic definitions and theorems about convex hulls in real vector
-  spaces.  TODO: prove inclusion in the class of subcomplex vector spaces.
+  In this section, we introduce real vector spaces.
 
 $)
 
+  $( Variant of ~ fvimacnv where membership of ` A ` in the domain is not
+     needed provided the containing class ` B ` does not contain the empty set.
+     Note that this antecedent would not be needed with definition ~ df-afv .
+     (Contributed by BJ, 7-Jan-2024.) $)
+  bj-fvimacnv0 $p |- ( ( Fun F /\ -. (/) e. B ) ->
+                                  ( ( F ` A ) e. B <-> A e. ( `' F " B ) ) ) $=
+    ( wfun c0 wcel wn wa cfv ccnv cima cdm wceq eleq1 biimpcd con3rr3 imp ndmfv
+    wi ex nsyl2 simpr fvimacnv biimpd com3l sylc com3r fvimacnvi adantr impbid
+    ) CDZEBFZGZHACIZBFZACJBKFZUKUMUOUPSZUMUOUKUPUMUOUKUPSZUMUOHZACLFZUOURUSUNEM
+    ZUTUMUOVAGUOVAULVAUOULUNEBNOPQACRUAUMUOUBUKUTUOUPUKUTUQUKUTHUOUPABCUCUDTUEU
+    FTUGQUKUPUOSUMUKUPUOABCUHTUIUJ $.
+
+  $( Fields are division rings.  (Contributed by BJ, 6-Jan-2024.) $)
+  bj-fielddivring $p |- Field C_ DivRing $=
+    ( cfield cdr ccrg cin df-field inss1 eqsstri ) ABCDBEBCFG $.
+
+  $( The field of real numbers is a division ring.  (Contributed by BJ,
+     6-Jan-2024.) $)
+  bj-rrdrg $p |- RRfld e. DivRing $=
+    ( cfield cdr crefld bj-fielddivring refld sselii ) ABCDEF $.
+
   $( Symbol for the class of real vector spaces. $)
   $c RRVec $.
 
   $( Syntax for the class of real vector spaces. $)
   crrvec $a class RRVec $.
 
-  $( Definition of the class of real vector spaces.  (Contributed by BJ,
-     9-Jun-2019.) $)
-  df-bj-rrvec $a |- RRVec = { x e. LVec | ( Scalar ` x ) = RRfld } $.
+  $( Definition of the class of real vector spaces.  The previous definition,
+     ` |- RRVec = { x e. LMod | ( Scalar `` x ) = RRfld } ` , can be recovered
+     using ~ bj-isrvec .  The present one is preferred since it does not use
```

Comment:
~ bj-isrvec does not exactly recover the previous definition: we would need the following theorem  for it:
`bj-isrvec0 $p |- ( V e. RRVec <-> ( V e. LVec  /\ ( Scalar ` V ) = RRfld ) )`
This can be proven by combining ~bj-isrvec with ~bj-rvecvec.

Reply:
You're right: actually I stated the (abusively called) "previous definition" using the class abstraction with `LMod` also, so bj-isrevc is essentially sufficient. I chose this less restrictive version, and then proved inclusion in `LVec` a few statements below.

Reply:
Nevertheless, I would like to have the corresponding theorem
` bj-isrvec0 $p |- ( V e. RRVec <-> ( V e. LVec  /\ ( Scalar V ) = RRfld ) )`
explicitly after ~bj-rvecvec, because it characzterizes the real vector spaces as vector spaces.
This can be done, however, in a separate PR.

Reply:
This is now #3752